src/pages/registration/FilesAndLicensePanel.tsx

```diff
@@ -121,370 +120,382 @@ export const FilesAndLicensePanel = ({ uppy }: FilesAndLicensePanelProps) => {
                         }
-                        data-testid={dataTestId.registrationWizard.files.linkToResourceField}
-                        onChange={(event) => {
-                          const inputValue = event.target.value;
-                          if (inputValue) {
-                            if (associatedLinkIndex < 0) {
-                              const newAssociatedLink: AssociatedLink = { type: 'AssociatedLink', id: inputValue };
-                              push(newAssociatedLink);
-                            } else {
-                              const fieldName = `${name}[${associatedLinkIndex}].${SpecificLinkFieldNames.Id}`;
-                              setFieldValue(fieldName, inputValue);
-                              setFieldTouched(fieldName);
-                            }
-                          } else {
-                            const associatedArtifactsBeforeRemoval = associatedArtifacts.length;
-                            remove(associatedLinkIndex);
-                            if (associatedArtifactsBeforeRemoval === 1) {
-                              // Ensure field is set to touched even if it's empty
-                              setFieldTouched(name);
-                            }
-                          }
-                        }}
-                      />
-                    )}
-                  </BackgroundDiv>
-                </Paper>
+                      } else {
+                        const fieldName = `${name}[${associatedLinkIndex}].${SpecificLinkFieldNames.Id}`;
+                        setFieldValue(fieldName, inputValue);
+                        setFieldTouched(fieldName);
+                      }
+                    } else {
+                      const associatedArtifactsBeforeRemoval = associatedArtifacts.length;
+                      remove(associatedLinkIndex);
+                      if (associatedArtifactsBeforeRemoval === 1) {
+                        // Ensure field is set to touched even if it's empty
+                        setFieldTouched(name);
+                      }
+                    }
+                  }}
+                />
+              )}
+            </Paper>
 
-                {associatedArtifacts.length === 0 &&
-                  typeof errors.associatedArtifacts === 'string' &&
-                  touched.associatedArtifacts && (
-                    <FormHelperText error>
-                      <ErrorMessage name={name} />
-                    </FormHelperText>
-                  )}
+            {(associatedArtifacts.length === 0 || isNullAssociatedArtifact) && !originalDoi && (
+              <Paper elevation={5} component={BackgroundDiv}>
+                <Typography variant="h2" paragraph>
+                  {t('registration.files_and_license.resource_is_a_reference')}
+                </Typography>
+                <Box sx={{ backgroundColor: 'white', width: '100%', p: '0.25rem 1rem' }}>
+                  <FormControlLabel
```

Comment:
Om denne er valgt nå, så kan man fortsatt laste opp filer. Og om man laster opp ei fil har man både `NullAssociatedArtifact` og `UnpublishedFile` som gjør at lagring returnerer 400 error.
2 mulige løsninger:
- Skjul filopplasting og lenke om denne checkboxen er valgt
- Fjern `NullAssociatedArtifact` om bruker legger til fil eller lenke

Ikke helt sikker selv på hva som blir mest logisk, men den første er nok enklere iaf. Ser det ikke er tegnet sånn i skissene, men jeg tipper dette er en case som kanskje ikke er tenkt på 🤔

Reply:
Syns også det virker veldig rart å ha en checkbox der, men man skal fortsatt kunne publisere uten filer og lenke, og uten å krysse av denne. Da har den jo ingen funksjon? jeg kan ta en prat med PO/designer om hva som er intensjonen her

Reply:
Snakka med PO nå, og det var visst en tapt drakamp mot eksterne ja :p 

En mulig løsning kan være noe sånt:
```
<FileUploader addFile={file => {
    if (hasNullArtifacte) {
      removeNullArtifact();
    }
    push(file);
  }} />
```
(veldig pseudokode dette da)
og samme for lenke på et vis

Reply:
Dobbeltsjekket: jeg klarer ikke å reprodusere at `NullAssociatedArtifact ` blir i tabellen når man legger til fil eller lenke.

Reply:
kan være rot fra meg også, skal sjekke igjen!